### PR TITLE
Update to support for open jdk17

### PR DESCRIPTION
**Pull request by jshughes**
#702 Update to Support for OpenJDK17

## 🗒️ Summary

The method LDDTool used for setting environment variables is no longer supported in Java17. Determined that there is no need to set the environment variables in the first place. (This was old code originally added by the developer of the Maven script.)

Changes:
Three of the four environment variables being set were no longer being used and so were deprecated.
The remaining variable, PARENT_DIR is now being set using System.getProperty("user.dir"); This is used to locate the /Data directory, were the config.properties and protege database files are put.
The offending code in AbstractGenerationPlugin.java was removed.

## ⚙️ Test Data and/or Report
The maven build ran successfully.

## ♻️ Related Issues
Will create new issue to add a command line option to set the relative location of the /Data directory line.

Resolves # 702

### CODE REVIEW

File: model-dmdocument/src/main/java/gov/nasa/pds/model/plugin/DMDocument.java [hunk 1]
```diff
@@ -456,9 +454,22 @@ public static void main(String args[]) throws Throwable {
     processArgumentParserNamespacePhase1(argparse4jNamespace);
 
     // first get the environment variables
-    getEnvMap();
-    dataDirPath = lPARENT_DIR + "/Data/";
-
+    String USER_DIR = System.getProperty("user.dir");
```
Comment (jordanpadams) [inline model-dmdocument/src/main/java/gov/nasa/pds/model/plugin/DMDocument.java:457]: @jshughes unfortunately, this doesn't work the way we want it to. By using the `user.dir` property, you are assuming the user is running the software from a specific location. However, if they are running it from their path or from some other directory, it won't work.

we probably want to do something like what [validate does here](https://github.com/NASA-PDS/validate/blob/main/src/main/resources/bin/validate#L66) where we set a new java property (`-Dresources.home=${PARENT_DIR}/resources`) and then we get that property in the launcher code (`System.getProperty("resources.home")`). So in this case you can add something like `-Ddata.home=${PARENT_DIR}/Data` to the LDDTool command-line script, and then pull that property in your code.

You can maybe use the `user.dir` as a fallback in the event `data.home` is not set.

hopefully that makes sense.

File: model-dmdocument/src/main/java/gov/nasa/pds/model/plugin/DMDocument.java [hunk 2]
```diff
@@ -456,9 +454,22 @@ public static void main(String args[]) throws Throwable {
...
-
+    String USER_DIR = System.getProperty("user.dir");
+    if (USER_DIR == null) {
+      registerMessage("3>error Environment variable USER_DIR is null");
+      printErrorMessages();
+      System.exit(1);
+    }
+    // 666
+//    System.out.println ("debug DMDocument -1- USER_DIR:" + USER_DIR);
+    USER_DIR = USER_DIR.replace('\\', '/');
```
Comment (jordanpadams) [inline model-dmdocument/src/main/java/gov/nasa/pds/model/plugin/DMDocument.java:465]: Also, I would rename this variable to use `camelCase`, which is more inline with common Java variable naming best practices. All caps are usually reserved for static environment variables, not local variables.

**Review by jordanpadams — CHANGES_REQUESTED** [review]

**Comment by jshughes** [comment]
@jordanpadams I just tried to push (actually re-push) my branch with the updates and got the following. Can a branch be re-pushed?
Issue742

**Comment by jordanpadams** [comment]
@jshughes i made some changes to the branch that looks like they were pulled. Can you try doing a git pull on your branch

**Comment by jshughes** [comment]
@jordanpadams In my local branch, Update_to_Support_for_OpenJDK17Try2, I did a Team -> Pull. A "Problem Occurred" with the following details. (Note: There are not a lot of changes so starting fresh with a new branch would not be a problem.)

Pulling 1 repository
Remote origin did not advertise Ref for branch refs/heads/Update_to_Support_for_OpenJDK17Try2. This Ref may not exist in the remote or may be hidden by permission settings.
Remote origin did not advertise Ref for branch refs/heads/Update_to_Support_for_OpenJDK17Try2. This Ref may not exist in the remote or may be hidden by permission settings.

**Comment by jordanpadams** [comment]
@jshughes can you create a new branch off of this one and I can take care of the rest?

**Pull request closed without merging**